### PR TITLE
Client Side Image Compression

### PR DESCRIPTION
**Pull request by alexyoung23j**
## Link to Issue
Closes: #4238 

## Description of Changes
- Adds a helper function `compressImage` that uses `browser-image-compression` to compress images to 10MB and max 1000px width or height (the constraints needed for Cloudflare Polish to work). This is used to compress files everywhere we upload to s3- if the file is not an image, it is not compressed and uploaded as normal.

## Test Plan
- Upload an image larger than 10MB or 1000px width/height and then visit the s3 url for the uploaded content, you should see the image has been compressed. 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

**Comment by jnaviask** [comment]
@alexyoung23j what were the reasons you ended up going with `browser-image-compression` vs an alternative library?

**Comment by CowMuon** [comment]
In general, "most of the articles I found were using it" is not exactly the best reason for using a library; but rather why is it that it is so widely used. Do we understand what makes it the best choice, or are we just blindly using it because everyone else is? 

While this is an important engineering principle that we follow, in the case of image resizing, I'm fine with this. However we do need to be surfacing more actual engineering in our engineering.

**Review by jnaviask — CHANGES_REQUESTED** [review]
How long do we expect compression to take? Do we show any sort of loading dialogue if it takes a long time?

**Comment by alexyoung23j** [comment]
@jnaviask In my tests I saw no noticeable difference in upload vs compression + upload time, but in the case of quill the UI already has a loading spinner.

File: yarn.lock [hunk 1]
```diff
@@ -21654,6 +21726,18 @@ quill@^1.3.4, quill@^1.3.6, quill@^1.3.7, quill@^2.0.0-dev.4:
     parchment "^1.1.4"
     quill-delta "^3.6.2"
 
+quill@^2.0.0-dev.4:
```
[the remark this answers is not in the document]
Comment (alexyoung23j) [reply yarn.lock:21729]: Hm, im looking now but not totally sure. I didn't touch that package (just added the browser compression package), and just ran yarn from the root. Seems like the version of quill we have in package.json is indeed this 2.0.0-dev.4, but previously was resolving to 1.3.7, and then running yarn changed that? @rbennettcw Do you have any insight about what's happening here?

[the remark this answers is not in the document]
Comment (rbennettcw) [reply yarn.lock:21729]: I don't think I've ever changed the `quill` dependency. Perhaps the addition of `react-quill` or one of the other quill-related deps required a different version of the base lib. I think the lock file shows not just the explicit deps, but also the indirect as well.

[the remark this answers is not in the document]
Comment (rbennettcw) [reply yarn.lock:21729]: CleanShot 2023-07-19 at 09 31 17@2x

Looks like `2.0.0…` has been there for a while though.

[the remark this answers is not in the document]
Comment (jnaviask) [reply yarn.lock:21729]: We can't just slide in a major version bump... let me try to regenerate the lock file properly.

[the remark this answers is not in the document]
Comment (jnaviask) [reply yarn.lock]: I just resolved it by downgrading the quill version manually -- we weren't on 2.x before, so we shouldn't specify it in the package.json.

**Review by jnaviask — APPROVED** [review]
Works.

**Comment by zakhap** [comment]
Screen Shot 2023-07-20 at 5 49 43 PM
Why does the uploading spinner show up behind the editor?

**Comment by CowMuon** [comment]
Looks like a regressing from downgrading Quill? @jnaviask @alexyoung23j

**Comment by CowMuon** [comment]
@alexyoung23j If this is merely a visual blip we would like to ship and the fix this as a fast follow. For that we need: the following: 

- [ ] Confirm this is a visual blemish, and not an affordance that does anything. (ie user cannot interact with it, no actual functionality is negatively or positively impacted.) 
- [ ] Estimate on time/points needed for fix. _**IF WE CAN DO THIS IN THE NEXT HOUR WE WILL HOLD DEPLOYMENT**_  or if it will take longer, we will ship as a fast follow @alexyoung23j 
- [ ] Confirm this was indeed caused by the commit that fixed the package.json <> yarn.lock conflict over Quill version. Our hypothesis is that was the commit that introduced this bug, we need to validate our hypothesis.

**Comment by alexyoung23j** [comment]
Confirmed that this doesn't affect functionality and is a visual blemish.

**Comment by CowMuon** [comment]
I would go ahead and bisect the youknowhat out of it, and post the bad commit here. Will circle back in 30 to decide whether to ship+FF or wait for fix.

**Comment by rbennettcw** [comment]
@CowMuon ^^ There's the fix in the mentioned PR

**Pull request merged**